### PR TITLE
Render error tx844

### PR DESCRIPTION
**Pull request by sohammk08**
## High Level Overview of Change

Made changes in translations.json to exclude _connection.server.publicKey_ from **server_ledgers_hint**

### Context of Change
This change was made because the _connection.server.publicKey_ in **server_ledgers_hint** in translations.json wasn't able to function properly. Upon closer inspection, it was found that the reason for this occurrence was the _publicKey_ field being empty. Removed the _connection.server.publicKey_ part and restructured sentences accordingly.

### Type of Change

- [x] Translation Updates

### TypeScript/Hooks Update

- [ ] Updated files to React Hooks
- [ ] Updated files to TypeScript

## Before / After
Before:
```json
"server_ledgers_hint": "This node {{connection.server.publicKey}} only contains ledgers {{connection.ledger.validated}}",
```
After:
```json
"server_ledgers_hint": "This node only contains ledgers {{connection.ledger.validated}}",
```

### CODE REVIEW

File: public/locales/en-US/translations.json [hunk 1]
```diff
@@ -200,7 +200,7 @@
   "invalid_transaction_hash": "The transaction hash is invalid",
   "ledger_not_found": "Ledger not Found",
   "check_ledger_id": "Please check your ledger id",
-  "server_ledgers_hint": "This node ({{connection.server.publicKey, truncate(length: 10)}}) only contains ledgers {{connection.ledger.validated}}",
```
Comment (mvadari) [inline public/locales/en-US/translations.json:203]: I think we still want the pubkey to be printed if it's available, with some sort of "This Clio node only..." alternate message

Comment (sohammk08) [reply public/locales/en-US/translations.json:203]: So, you're saying that the **clio-node-part** still needs to be present in the code. What if I change the code such that whenever the **pubKey** is not empty, it gets displayed. But when it is empty, print it as "This clio node only..."

Would that be viable?

Comment (mvadari) [reply public/locales/en-US/translations.json:203]: Yes, something like that sounds good.

**Comment by sohammk08** [comment]
@ckniffen @pdp2121 @mvadari 
I finally have an update. I managed to get the publicKey part working.

However, I'm still not sure about the part where we display **server_ledgers_hint** if publicKey does not exist and **server_ledgers_hint_with_key** if it does.

Here is where I currently am, for reference:

Capture

**Comment by pdp2121** [comment]
> @ckniffen @pdp2121 @mvadari I finally have an update. I managed to get the publicKey part working.
> 
> However, I'm still not sure about the part where we display **server_ledgers_hint** if publicKey does not exist and **server_ledgers_hint_with_key** if it does.
> 
> Here is where I currently am, for reference:
> 
> Capture

@sohammk08  You could use ternary operator to condition on whether the publicKey exists and display different messages based on that.

**Comment by sohammk08** [comment]
@pdp2121 @mvadari @ckniffen I have solved the error and now, it works properly. Just to make sure, I have also tested to see if it works, and it does. I am giving below the method I used to test. I will be making a commit containing the changes soon after writing this comment. Below is how I tested it to make sure if it works or not. Take a look:

Original file:
```typescript
  const notFound = title.includes('not_found')
  const hintMsg = hints.map((hint) => (
    <div className="hint" key={hint}>
      {t(hint as any, values)}
    </div>
  ))
  const derivedWarning = warning ?? (notFound && t('not_found'))

  return (
    <div className="no-match">
      <Helmet title={t(title as any)} />
      {isError && <div className="uh-oh">{t('uh_oh')}</div>}
      <div className="title">{t(title as any, values)}</div>
      {hintMsg}
      {(derivedWarning || isError) && (
        <div className="warning">
          <InfoIcon title={derivedWarning} />
          &nbsp;
          <span>{derivedWarning}</span>
        </div>
      )}
    </div>
  )
}

export default NoMatch
```

Changed file:
``` typescript
  const notFound = title.includes('not_found')

  // Determine which hint message to display based on the presence of publicKey
  const hintMsg = values.connection?.server?.publicKey
    ? t('server_ledgers_hint_with_key', values)
    : t('server_ledgers_hint', values)

  const derivedWarning = warning ?? (notFound && t('not_found'))

  return (
    <div className="no-match">
      <Helmet title={t(title as any)} />
      {isError && <div className="uh-oh">{t('uh_oh')}</div>}
      <div className="title">{t(title as any, values)}</div>
      {hintMsg && <div className="hint">{hintMsg}</div>}
      {(derivedWarning || isError) && (
        <div className="warning">
          <InfoIcon title={derivedWarning} />
          &nbsp;
          <span>{derivedWarning}</span>
        </div>
      )}
    </div>
  )
}

export default NoMatch
```

I also made changes to translations.json as such:
``` json
  "server_ledgers_hint": "This node only contains ledgers {{connection.ledger.validated}}",
  "server_ledgers_hint_with_key": "This node ({{connection.server.publicKey, truncate(length: 10)}}) only contains ledgers {{connection.ledger.validated}}",
```

Also, here is how I tested whether the change works or not;
First, I set the publicKey to null for debugging purposes. Then, I changed the hintMsg line. Here is how it looks in code:
```typescript
  const notFound = title.includes('not_found')
  const publicKey = null

  // Determine which hint message to display based on the presence of publicKey
  const hintMsg = publicKey
    ? t('server_ledgers_hint_with_key', values)
    : t('server_ledgers_hint', values)

  const derivedWarning = warning ?? (notFound && t('not_found'))
```

I checked the logic after adding the above debugging code as said before. Also, please let me know if I should try and implement a test in NoMatch.test.tsx for the same.

Have a good day :)

**Comment by pdp2121** [comment]
> @pdp2121 @mvadari @ckniffen I have solved the error and now, it works properly. Just to make sure, I have also tested to see if it works, and it does. I am giving below the method I used to test. I will be making a commit containing the changes soon after writing this comment. Below is how I tested it to make sure if it works or not. Take a look:
> 
> Original file:
> 
> ```ts
>   const notFound = title.includes('not_found')
>   const hintMsg = hints.map((hint) => (
>     <div className="hint" key={hint}>
>       {t(hint as any, values)}
>     </div>
>   ))
>   const derivedWarning = warning ?? (notFound && t('not_found'))
> 
>   return (
>     <div className="no-match">
>       <Helmet title={t(title as any)} />
>       {isError && <div className="uh-oh">{t('uh_oh')}</div>}
>       <div className="title">{t(title as any, values)}</div>
>       {hintMsg}
>       {(derivedWarning || isError) && (
>         <div className="warning">
>           <InfoIcon title={derivedWarning} />
>           &nbsp;
>           <span>{derivedWarning}</span>
>         </div>
>       )}
>     </div>
>   )
> }
> 
> export default NoMatch
> ```
> 
> Changed file:
> 
> ```ts
>   const notFound = title.includes('not_found')
> 
>   // Determine which hint message to display based on the presence of publicKey
>   const hintMsg = values.connection?.server?.publicKey
>     ? t('server_ledgers_hint_with_key', values)
>     : t('server_ledgers_hint', values)
> 
>   const derivedWarning = warning ?? (notFound && t('not_found'))
> 
>   return (
>     <div className="no-match">
>       <Helmet title={t(title as any)} />
>       {isError && <div className="uh-oh">{t('uh_oh')}</div>}
>       <div className="title">{t(title as any, values)}</div>
>       {hintMsg && <div className="hint">{hintMsg}</div>}
>       {(derivedWarning || isError) && (
>         <div className="warning">
>           <InfoIcon title={derivedWarning} />
>           &nbsp;
>           <span>{derivedWarning}</span>
>         </div>
>       )}
>     </div>
>   )
> }
> 
> export default NoMatch
> ```
> 
> I also made changes to translations.json as such:
> 
> ```json
>   "server_ledgers_hint": "This node only contains ledgers {{connection.ledger.validated}}",
>   "server_ledgers_hint_with_key": "This node ({{connection.server.publicKey, truncate(length: 10)}}) only contains ledgers {{connection.ledger.validated}}",
> ```
> 
> Also, here is how I tested whether the change works or not; First, I set the publicKey to null for debugging purposes. Then, I changed the hintMsg line. Here is how it looks in code:
> 
> ```ts
>   const notFound = title.includes('not_found')
>   const publicKey = null
> 
>   // Determine which hint message to display based on the presence of publicKey
>   const hintMsg = publicKey
>     ? t('server_ledgers_hint_with_key', values)
>     : t('server_ledgers_hint', values)
> 
>   const derivedWarning = warning ?? (notFound && t('not_found'))
> ```
> 
> I checked the logic after adding the above debugging code as said before. Please let me know if I should try and implement a test in NoMatch.test.tsx for the same. Also, as @pdp2121 suggested, I set the server_ledgers_hint_with_key for other language translations as _null_
> 
> Have a good day :)

Hi @sohammk08, it would be nice if you could add a test inside NoMatch.test.tsx. Thanks for your contribution!

File: src/containers/NoMatch/index.tsx [hunk 2]
```diff
@@ -40,19 +40,20 @@ const NoMatch = ({
...
+
   const derivedWarning = warning ?? (notFound && t('not_found'))
 
   return (
     <div className="no-match">
       <Helmet title={t(title as any)} />
       {isError && <div className="uh-oh">{t('uh_oh')}</div>}
       <div className="title">{t(title as any, values)}</div>
-      {hintMsg}
+      {hintMsg && <div className="hint">{hintMsg}</div>}
```
Comment (pdp2121) [inline src/containers/NoMatch/index.tsx:56]: I'm not sure why we need to change here. Would `hintMsg` ever be `null`? Also, are we not using `hints` array here at all?

Comment (sohammk08) [reply src/containers/NoMatch/index.tsx:56]: Correct, at present, the _hints_ array isn't directly used within the JSX rendering logic of the component.
Instead, it's utilized within the _useEffect_ hook to concatenate its values into the _description_ parameter of the _track_ function.

**Comment by sohammk08** [comment]
> Tests are failing since we are using node 18 instead.

Yes. I checked the staging branch of the ripple/staging and all tests are passing. I ran _npm run test_ and almost 11 tests are failing. Hopefully, I will be committing the revised code such that it doesn't fail any tests and.
Thanks for your input.

Good Day.
